harvard-university-for-the-creative-arts.csl

```diff
@@ -110,20 +131,30 @@
           </else-if>
         </choose>
       </if>
-      <else-if type="manuscript song post-weblog post personal_communication speech" match="any">
-        <group delimiter=". ">
+      <else-if type="manuscript song post-weblog post" match="any">
+        <group delimiter=" ">
           <text variable="genre" prefix="[" suffix="]"/>
           <text variable="medium" prefix="[" suffix="]"/>
-          <text variable="note" prefix="[" suffix="]"/>
         </group>
         <text variable="dimensions" prefix=" " suffix="."/>
       </else-if>
       <else-if type="broadcast"/>
       <else-if type="graphic" match="any">
-        <group>
+        <group delimiter=" ">
           <text variable="genre"/>
           <text variable="medium" prefix="[" suffix="]"/>
+          <text variable="dimensions" suffix="."/>
           <text variable="note" prefix=" " suffix="."/>
+          <text variable="archive_location" prefix=" "/>
+        </group>
+        <text variable="archive" prefix=": " suffix="."/>
+      </else-if>
+      <else-if type="speech personal_communication" match="any">
+        <group delimiter=" " prefix="[" suffix="]">
+          <text variable="genre" suffix=" "/>
+          <text variable="note"/>
```

Comment:
@ibadger, do you still need "note" for "speech personal_communication" and "graphic" (above)?